syntax = "proto3";

package pipeline;

// Event sent from Module A to Module B
message Event {
  string id = 1;
  string data = 2;
}

// Text sent from Module B to Module C
message Comment {
  string id = 1;
  string text = 2;
}

// Audio sent from Module C to Module D
message Audio {
  string id = 1;
  bytes audio_data = 2;
}

// Generic response used by all services
message BasicResponse {
  string id = 1;
  bool success = 2;
  string message = 3;
}

// Service run by Module B
service ModuleB {
  rpc ProcessEvent(Event) returns (BasicResponse);
}

// Service run by Module C
service ModuleC {
  rpc TextToSpeech(Comment) returns (BasicResponse);
}

// Service run by Module D
service ModuleD {
  rpc PlayAudio(Audio) returns (BasicResponse);
} 